fast/stages/2-networking-a-peering/dns-landing.tf

```diff
@@ -50,82 +78,43 @@ module "gcp-example-dns-private-zone" {
-  type            = "private"
-  name            = "gcr-io"
-  domain          = "gcr.io."
-  client_networks = [module.landing-vpc.self_link]
-  recordsets = {
-    "A gcr.io." = { ttl = 300, records = [
-      "199.36.153.8", "199.36.153.9", "199.36.153.10", "199.36.153.11"
-    ] }
-    "CNAME *" = { ttl = 300, records = ["private.googleapis.com."] }
-  }
-}
-
-module "packages-private-zone" {
-  source          = "../../../modules/dns"
-  project_id      = module.landing-project.project_id
-  type            = "private"
-  name            = "packages-cloud"
-  domain          = "packages.cloud.google.com."
-  client_networks = [module.landing-vpc.self_link]
-  recordsets = {
-    "A packages.cloud.google.com." = { ttl = 300, records = [
-      "199.36.153.8", "199.36.153.9", "199.36.153.10", "199.36.153.11"
-    ] }
-    "CNAME *" = { ttl = 300, records = ["private.googleapis.com."] }
-  }
-}
+# Google APIs via response policies
 
-module "pkgdev-private-zone" {
-  source          = "../../../modules/dns"
-  project_id      = module.landing-project.project_id
-  type            = "private"
-  name            = "pkg-dev"
-  domain          = "pkg.dev."
-  client_networks = [module.landing-vpc.self_link]
-  recordsets = {
-    "A pkg.dev." = { ttl = 300, records = [
-      "199.36.153.8", "199.36.153.9", "199.36.153.10", "199.36.153.11"
-    ] }
-    "CNAME *" = { ttl = 300, records = ["private.googleapis.com."] }
-  }
-}
-
-module "pkigoog-private-zone" {
-  source          = "../../../modules/dns"
-  project_id      = module.landing-project.project_id
-  type            = "private"
-  name            = "pki-goog"
-  domain          = "pki.goog."
-  client_networks = [module.landing-vpc.self_link]
-  recordsets = {
-    "A pki.goog." = { ttl = 300, records = [
-      "199.36.153.8", "199.36.153.9", "199.36.153.10", "199.36.153.11"
-    ] }
-    "CNAME *" = { ttl = 300, records = ["private.googleapis.com."] }
+module "landing-dns-policy-googleapis" {
+  source     = "../../../modules/dns-response-policy"
+  project_id = module.landing-project.project_id
+  name       = "googleapis"
+  networks = {
+    landing = module.landing-vpc.self_link
   }
+  rules = merge(
+    {
+      googleapis-all = {
+        dns_name = "*.googleapis.com."
+        local_data = { CNAME = { rrdatas = [
+          "private.googleapis.com."
+        ] } }
+      }
+      googleapis-private = {
+        dns_name = "private.googleapis.com."
+        local_data = { A = { rrdatas = [
+          "199.36.153.8", "199.36.153.9", "199.36.153.10", "199.36.153.11"
+        ] } }
+      }
+      googleapis-restricted = {
+        dns_name = "restricted.googleapis.com."
+        local_data = { A = { rrdatas = [
+          "199.36.153.4", "199.36.153.5", "199.36.153.6", "199.36.153.7"
+        ] } }
+      }
+    },
+    {
+      for k, v in local.googleapis_domains : k => {
+        dns_name = v
+        local_data = { CNAME = { rrdatas = [
```

Comment:
Will the naked domains (like "gcr.io") work with CNAMEs? I don't think so. (They usually require explicit pointing to the IPs).

Reply:
let me try... they seem to work

```
ext_ludomagno_google_com@bastion:~$ dig pkg.dev +short
private.googleapis.com.
199.36.153.8
199.36.153.9
199.36.153.10
199.36.153.11
```